src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -68,6 +70,14 @@ class AuthScreens extends React.Component {
             authEndpoint: `${CONFIG.EXPENSIFY.URL_API_ROOT}api?command=Push_Authenticate`,
         }).then(subscribeToReportCommentEvents);
 
+        // If we have a validation code in our URL ensure we validate the associated login
+        if (this.props.currentURL.includes(`/${ROUTES.VALIDATE_LOGIN}/`)) {
+            // Split the /v/accountID/validateCode URL string into distinct components
+            // "", "v", "accountID", "validateCode"
+            const urlArray = this.props.currentURL.split('/');
+            User.validateLogin(Number(urlArray[2]), urlArray[3]);
```

Comment:
It looks like this check is here so that we can handle the case of a user that is already authenticated. But if a user is already authenticated then presumably they are not logged in with the login they are trying to validate. 

Will weird stuff happen if we just validate this login anyway? e.g. do we need to clear out any stored data? Maybe force a sign out and then validate the login? Or something else? I haven't tested this yet so not sure what will happen but definitely curious.

Reply:
I'm not sure I understand the flow you're describing here. This flow is for validating secondary logins so it's not possible for the user to be logged in with the login they're validating at this point.

Reply:
The flow would be...

1. User A logs in with primary account
2. User A creates a separate account (Note: Not secondary login) and gets a validate link for that
3. Since User A is logged in with the first account they created we will validate the account from the link and then change their auth token.

Maybe not a common occurrence, but I think it's weird that we can even do it. My expectation would be:

User A gets cleanly logged out of the first account they created and gets logged into the second account.

But maybe I'm misunderstanding this and that's not actually possible.

Reply:
For validating primary accounts we use the `/setpassword/` URL so I think we would be fine, I will try testing this flow and try some edgecases for secondary logins to see what happens